src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder.cs

```diff
@@ -840,5 +699,50 @@ bool metadataTypeMatches(SymbolSet s, INamedTypeSymbol t)
 
         private static SymbolSet CreateSymbolAndProjectIdSet()
             => s_setPool.Allocate();
+
+        private class KeyEqualityComparer : IEqualityComparer<(SymbolKey, ProjectId, IImmutableSet<Project>)>
```

Comment:
will doc this.  but the effective idea is to properly support equals/gethashcode given that IImmutableSet doesn't automatically give you that.